### PR TITLE
feat: Add configurable batch_size and max_workers to embed method

### LINKED ISSUE #534: Allow users to configure `embed_batch_size` or `ThreadPoolExecutor` size when calling `Client.embed`
It looks like batching was added in #437 - thank you for implementing this, it's very helpful.

I notice that batching, as defined [here](https://github.com/cohere-ai/cohere-python/blob/main/src/cohere/client.py#L135), depends on a [fixed batch size](https://github.com/cohere-ai/cohere-python/blob/main/src/cohere/config.py#L1). This can be suboptimal for clients submitting a large number of smaller documents, as we cannot [configure the ThreadPoolExecutor size](https://github.com/cohere-ai/cohere-python/blob/main/src/cohere/client.py#L111) to parallelize a large number of small data payloads. As a result a client might end up blocking while waiting for small network responses.

Would it be possible to allow clients to configure either the `ThreadPoolExecutor` size or the `embed_batch_size` setting when calling `embed`?

### PR DESCRIPTION
**Pull request by fede-kamel**
# Add configurable batch_size and max_workers to embed method

## Summary
This PR fixes #534 by making the embed batch size configurable through optional parameters, giving users control over batching behavior based on their specific needs.

## Problem
Previously, the `embed()` method used a fixed batch size of 96 (from `config.embed_batch_size`), which could be suboptimal for various use cases:
- Users with memory constraints needed smaller batches
- Users with high-throughput needs wanted larger batches
- Rate-limited applications needed to control concurrency

## Solution
Added two optional parameters to the `embed()` method:
- `batch_size: Optional[int] = None` - Controls the number of texts per batch
- `max_workers: Optional[int] = None` - Controls ThreadPoolExecutor concurrency (sync client only)

### Changes to `src/cohere/client.py`:
```python
def embed(
    self,
    *,
    texts: Optional[Sequence[str]] = OMIT,
    # ... existing parameters ...
    batch_size: Optional[int] = None,  # NEW
    max_workers: Optional[int] = None,  # NEW
) -> EmbedResponse:
```

The implementation:
1. Uses provided `batch_size` or falls back to the default `embed_batch_size` (96)
2. Creates a temporary ThreadPoolExecutor if `max_workers` is specified
3. Maintains full backward compatibility - existing code continues to work unchanged

### All tests pass:
```bash
$ python -m pytest tests/test_configurable_batch_size.py -v
============================= test session starts ==============================
collected 6 items

tests/test_configurable_batch_size.py::TestConfigurableBatchSize::test_batch_size_edge_cases PASSED [ 16%]
tests/test_configurable_batch_size.py::TestConfigurableBatchSize::test_custom_batch_size PASSED [ 33%]
tests/test_configurable_batch_size.py::TestConfigurableBatchSize::test_custom_max_workers PASSED [ 50%]
tests/test_configurable_batch_size.py::TestConfigurableBatchSize::test_default_batch_size PASSED [ 66%]
tests/test_configurable_batch_size.py::TestConfigurableBatchSize::test_no_batching_ignores_parameters PASSED [ 83%]
tests/test_configurable_batch_size.py::TestAsyncConfigurableBatchSize::test_async_custom_batch_size PASSED [100%]

============================== 6 passed in 0.40s ===============================
```

### Test coverage includes:
- ✅ Custom batch sizes work correctly
- ✅ Default batch size (96) is used when parameter not specified
- ✅ Edge cases: batch_size=1, batch_size > total texts
- ✅ Custom max_workers creates new ThreadPoolExecutor
- ✅ Parameters are properly ignored when batching=False
- ✅ Async client batch_size support

## Code Quality
- ✅ Ruff linting passes
- ✅ Mypy type checking passes
- ✅ Import ordering fixed automatically by ruff

### Default behavior (unchanged):
```python
response = client.embed(texts=texts, model="embed-english-v3.0")
# Uses default batch_size=96
```

### Custom batch size for memory optimization:
```python
response = client.embed(
    texts=texts,
    model="embed-english-v3.0", 
    batch_size=10  # Smaller batches for memory-constrained environments
)
```

### Rate limiting with reduced concurrency:
```python
response = client.embed(
    texts=texts,
    model="embed-english-v3.0",
    batch_size=20,
    max_workers=2  # Only 2 concurrent API calls
)
```

## Benefits
1. **Memory optimization**: Users can reduce batch size to limit memory usage
2. **Performance tuning**: Users can increase batch size for fewer API calls
3. **Rate limit handling**: Control concurrency with max_workers
4. **Backward compatible**: No changes required to existing code
5. **Complements PR #698**: Works well with the memory-efficient `embed_stream()` method

This implementation provides the flexibility requested in issue #534 while maintaining the SDK's ease of use and backward compatibility.

---

> [!NOTE]
> **Medium Risk**
> Touches core `embed()` batching/concurrency behavior and introduces new parsing logic for streamed embeddings, which could affect performance and ordering/indexing edge cases (duplicates, multi-type embeddings), though changes are additive and covered by tests.
> 
> **Overview**
> Adds **configurable batching controls** to `Client.embed()` and `AsyncClient.embed()` via new optional `batch_size` (override default `embed_batch_size`) and `max_workers` (sync-only, via a temporary `ThreadPoolExecutor`; async warns and ignores).
> 
> Introduces **streaming embedding APIs**: `BaseCohere.embed_stream()` and `V2Client.embed_stream()` yield embeddings incrementally in batches, backed by new `streaming_utils.py` (`StreamedEmbedding`, `StreamingEmbedParser`) which uses `ijson` when available and falls back to full JSON parsing. Adds unit tests covering batch sizing/concurrency and streaming behavior, and updates `.gitignore` to ignore `.venv/`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d2d3d6575899fde44051f37701bfd48367003c8a. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/cohere/v2/client.py [hunk 1]
```diff
@@ -489,6 +489,128 @@ def embed(
...
+            )
+            
+            # Parse embeddings from response incrementally
+            parser = StreamingEmbedParser(response._response, batch_texts)
+            for embedding in parser.iter_embeddings():
+                # The parser sets embedding.text correctly for multiple embedding types
+                # Adjust the global index based on text position in batch
+                if embedding.text and embedding.text in batch_texts:
+                    text_idx_in_batch = batch_texts.index(embedding.text)
+                    embedding.index = batch_start + text_idx_in_batch
```
[the remark this answers is not in the document]
Comment (fede-kamel) [reply src/cohere/v2/client.py]: Fixed - added `used_batch_indices` tracking to match the base_client implementation:

```python
used_batch_indices: set[int] = set()
for embedding in parser.iter_embeddings():
    if embedding.text and embedding.text in batch_texts:
        for idx, text in enumerate(batch_texts):
            if text == embedding.text and idx not in used_batch_indices:
                text_idx_in_batch = idx
                used_batch_indices.add(idx)
                break
```

Now duplicate texts get unique indices.

File: src/cohere/base_client.py [hunk 2]
```diff
@@ -1125,6 +1125,123 @@ def embed(
...
+        if texts is None or texts is OMIT:
+            return
+        if not texts:
+            return
+
+        from .streaming_utils import StreamingEmbedParser
+
+        # Process texts in batches
+        texts_list = list(texts)
+        total_embeddings_yielded = 0
```
[the remark this answers is not in the document]
Comment (fede-kamel) [reply src/cohere/base_client.py]: Fixed - removed the unused `total_embeddings_yielded` variable from both `base_client.py` and `v2/client.py`.
